sdks/go/pkg/beam/core/runtime/harness/datamgr_test.go

```diff
@@ -0,0 +1,89 @@
+package harness
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"log"
+	"testing"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/exec"
+	pb "github.com/apache/beam/sdks/go/pkg/beam/model/fnexecution_v1"
+)
+
+type fakeClient struct {
+	t     *testing.T
+	done  chan bool
+	calls int
+}
+
+func (f *fakeClient) Recv() (*pb.Elements, error) {
+	f.calls++
+	data := []byte{1, 2, 3, 4}
+	elemData := pb.Elements_Data{
+		InstructionReference: "inst_ref",
+		Data:                 data,
+		Target: &pb.Target{
+			PrimitiveTransformReference: "ptr",
+			Name: "instruction_name",
+		},
+	}
+
+	msg := pb.Elements{}
+
+	for i := 0; i < bufElements+1; i++ {
+		msg.Data = append(msg.Data, &elemData)
+	}
+
+	// The first two calls fill up the buffer completely to stimulate the deadlock
```

Comment:
<!--thread_id:cc_183581014_t; commit:c99d4fe17f53af898950a67b6b7d816c570ad9aa; resolved:1-->
<!--section:context-quote-->
> **wcn3** wrote:
> Stimulate is more properly correct here, since I'm creating the actual condition of the deadlock. It's not a simulation of a deadlock; without the fix, it truly deadlocks. :)
> 
> If the word is confusing due to similarity, happy to replace with 'to force the deadlock'

<!--section:body-->
I could have justified either way, so it was largely a typo check. Thanks for the clarification!